app/models/container_image.rb

```diff
@@ -7,6 +7,7 @@ class ContainerImage < ActiveRecord::Base
   belongs_to :container_image_registry
   belongs_to :ext_management_system, :foreign_key => "ems_id"
   has_many :containers
+  has_many :container_nodes, -> { distinct }, :through => :containers
```

Comment:
@moolitayer please add a test for this.

Use `FactoryGirl` to create two (or more) containers with the same image and located on the same node. Then check that the `container_nodes` has no duplicated entries.